reader_test.go

```diff
@@ -296,6 +296,167 @@ func TestDecodeMediaPlaylistAutoDetectExtend(t *testing.T) {
+// m3u8.FullTimeParse()
+func TestTimeLayoutsDecode(t *testing.T) {
+	time_in_utc := "#EXT-X-PROGRAM-DATE-TIME:2006-01-02T15:04:05Z"
+	time_in_utc_nano := "#EXT-X-PROGRAM-DATE-TIME:2006-01-02T15:04:05.123456789Z"
+	time_with_positive_zone_and_colon := "#EXT-X-PROGRAM-DATE-TIME:2006-01-02T15:04:05+01:00"
+	time_with_positive_zone_no_colon := "#EXT-X-PROGRAM-DATE-TIME:2006-01-02T15:04:05+0100"
+	time_with_positive_zone_2digits := "#EXT-X-PROGRAM-DATE-TIME:2006-01-02T15:04:05+01"
+	time_with_negative_zone_and_colon := "#EXT-X-PROGRAM-DATE-TIME:2006-01-02T15:04:05-01:00"
+	time_with_negative_zone_no_colon := "#EXT-X-PROGRAM-DATE-TIME:2006-01-02T15:04:05-0100"
+	time_with_negative_zone_2digits := "#EXT-X-PROGRAM-DATE-TIME:2006-01-02T15:04:05-01"
+
+	p, e := NewMediaPlaylist(1, 2)
+	if e != nil {
+		t.Fatalf("Create media playlist failed: %s", e)
+	}
+	state := new(decodingState)
+	wv := new(WV)
+
+	var err error
+	err = decodeLineOfMediaPlaylist(p, wv, state, time_in_utc, true)
+	if err != nil {
+		t.Errorf("Time Parser Error for %s: %s", time_in_utc, err)
+	}
+
+	state = new(decodingState)
+	wv = new(WV)
+	err = decodeLineOfMediaPlaylist(p, wv, state, time_in_utc_nano, true)
+	if err != nil {
+		t.Errorf("Time Parser Error for %s: %s", time_in_utc_nano, err)
+	}
+
+	state = new(decodingState)
+	wv = new(WV)
+	err = decodeLineOfMediaPlaylist(p, wv, state, time_with_positive_zone_and_colon, true)
+	if err != nil {
+		t.Errorf("Time Parser Error for %s: %s", time_with_positive_zone_and_colon, err)
+	}
+
+	state = new(decodingState)
+	wv = new(WV)
+	err = decodeLineOfMediaPlaylist(p, wv, state, time_with_positive_zone_no_colon, true)
+	if err != nil {
+		t.Errorf("Time Parser Error for %s: %s", time_with_positive_zone_no_colon, err)
+	}
+
+	state = new(decodingState)
+	wv = new(WV)
+	err = decodeLineOfMediaPlaylist(p, wv, state, time_with_positive_zone_2digits, true)
+	if err != nil {
+		t.Errorf("Time Parser Error for %s: %s", time_with_positive_zone_2digits, err)
+	}
+
+	state = new(decodingState)
+	wv = new(WV)
+	err = decodeLineOfMediaPlaylist(p, wv, state, time_with_negative_zone_and_colon, true)
+	if err != nil {
+		t.Errorf("Time Parser Error for %s: %s", time_with_negative_zone_and_colon, err)
+	}
+
+	state = new(decodingState)
+	wv = new(WV)
+	err = decodeLineOfMediaPlaylist(p, wv, state, time_with_negative_zone_no_colon, true)
+	if err != nil {
+		t.Errorf("Time Parser Error for %s: %s", time_with_negative_zone_no_colon, err)
+	}
+
+	state = new(decodingState)
+	wv = new(WV)
+	err = decodeLineOfMediaPlaylist(p, wv, state, time_with_negative_zone_2digits, true)
+	if err != nil {
+		t.Errorf("Time Parser Error for %s: %s", time_with_negative_zone_2digits, err)
+	}
+}
+
+// Tests for Time Parsing of EXT-X-PROGRAM-DATE-TIME
+// We testing Strict format of RFC3339 where we can get time in UTC, UTC with Nanoseconds
+// timeZone in formats '±00:00', '±0000', '±00'
+// m3u8.StrictTimeParse()
+func TestStrictTimeLayoutsDecode(t *testing.T) {
```

Comment:
This test is really just testing the `StrictTimeParse` function, but it's testing it via decodeLineOfMediaPlaylist, I'd just call this function directly.

Also, there's a lot of repetition in the tests, can you rewrite using https://github.com/golang/go/wiki/TableDrivenTests (or sub tests) similar to https://github.com/grafov/m3u8/blob/f55aa7a0ae5bfe63e82ee5adefa3aa4e8692b3dd/writer_test.go#L223-L253 ?